pandas/_libs/tslib.pyx

```diff
@@ -666,6 +668,16 @@ cdef _array_to_datetime_object(
 
         cnp.PyArray_MultiIter_NEXT(mi)
 
+    if len(unique_timezones) > 1:
```

Comment:
is this condition ever False?

Reply:
yup pandas/tests/tools/test_to_datetime.py::test_to_datetime_with_empty_str_utc_false_format_mixed